### PR TITLE
docs(README): update README for the newest changes

### PR DESCRIPTION
**Pull request by nicoprow**
* Now separated into Pool and Gate
* Reflects the new app configuration without mentioning environments
* Added information about Github Workflows

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -3,284 +3,414 @@
...
+==== Pool-Auth
+
+On default configuration, the BPDM Gate expects the API of the BPDM Pool to be accessible without authorization requirements.
+In case the Pool instance to connect to have authorization activated, you need to activate this profile.
+The file `application-pool-auth.properties` configures the oAuth2 client for connecting to a secured BPDM Pool.
+Per default, the client will try to acquire a token via client credentials flow and expects the environment variable `BPDM_KEYCLOAK_SECRET` to contain the secret for the client.
+
+=== Helm Deployment
+
+This repository contains a Helm files for deploying the BPDM Gate to a Kubernetes environment.
```
Comment (pazepaze) [inline README.adoc:282]: a helm chart?

File: README.adoc [hunk 2]
```diff
@@ -3,284 +3,414 @@
...
+  - pool-auth
 ----
 
-As with application secrets instead of writing your credentials directly into a value.yaml you better pass them via command line when deploying the helm release: `--set imagePullSecret.user=your_user`
+==== Ingress
+
+You can specify your own ingress configuration for the Helm deployment to make the BPDM Gate available over Ingress.
+Note that you need to have the appropriate Ingress controller installed in your cluster first.
+For example, consider a Kubernetes cluster with an https://kubernetes.github.io/ingress-nginx/[Ingress-Nginx] installed.
+An Ingress configuration for the GAte deployment could look like this:
```
Comment (pazepaze) [inline README.adoc:353]: Gate

File: README.adoc [hunk 3]
```diff
@@ -3,284 +3,414 @@
...
+        - path: /
+          pathType: Prefix
+----
+
+==== Gate Configuration
+
+For the default deployment you already need to overwrite the configuration properties of the application.
+The Helm deployment comes with the ability to configure the BPDM Gate application directly over the values file.
+This way you are able to overwrite any configuration property of the `application.properties`,  `application-auth.properties` and  `application-pool-auth.properties` files.
+Consider that you would need to turn on `auth` and `pool-uath` profile first before overwriting any property in the corresponding properties file could take effect.
```
Comment (pazepaze) [inline README.adoc:374]: pool-auth

**Review by pazepaze — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 4]
```diff
@@ -3,284 +3,414 @@
...
+== BPDM Pool
 
-[%header,cols="h,1,1"]
-|===
-|
-|Description
-|Link
+The BPDM Pool is the single source of truth in Catena-X for business partner base information such as addresses and official identifiers.
+Each record in the Pool has a unique identifier with which it can be referenced across the entire Catena-X landscape, the business partner number.
+Business partner records are divided into legal entities, sites and partner addresses, where a legal entity may have several sites and a site several partner addresses.
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc]: rephrased this part

File: README.adoc [hunk 5]
```diff
@@ -3,284 +3,414 @@
...
-Each of the application properties files are multi-documents containing default values for their component as well as value overrides for different environments.
-Currently, the application can be run in the following environment profiles:
-
-* default: when no environment specified
-* local: for the execution on local machine (developer profile)
-* dev: for execution in remote dev environment
-* int: for execution in remote int environment
+The default configuration of the application is determined by the `application.properties` file.
+Here you can find core application configuration such as Swagger documentation, BPN generation and Actuator.
+Furthermore, you can find here the configuration for the connection to the Spring datasource (currently, developed against PostgreSQL) and Opensearch.
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc]: fixed

File: README.adoc [hunk 6]
```diff
@@ -3,284 +3,414 @@
...
+`application-auth.properties` enables authorization of endpoints and configures the connection to a Keycloak instance on which the authorization relies on.
+The application expects the Keycloak to run on `localhost` on port `8180` and needs a client secret has to be submitted via environment variable `BPDM_KEYCLOAK_SECRET`.
+However, as with the Spring datasource and Opensearch connections, the connection to the Keycloak can be freely configured.
 Besides the URL of the Keycloak the configuration determines the client credentials to validate incoming tokens (Standard behaviour of this application is bearer-only).
-The client secret has to be submitted via environment variable `BPDM_KEYCLOAK_SECRET`.
+
+For authorization purposes the application checks incoming token's permissions:
+
+. add_company_data: For endpoints creating or updating business partner records including triggering imports from CDQ/exports to Opensearch
+. view_company_data: For read-only endpoints of business partner data
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc]: changed the list formatting

File: README.adoc [hunk 7]
```diff
@@ -3,284 +3,414 @@
...
+
+The BPDM Pool looks for these permissions in the client/resource and not on the realm level.
 
 This profile also enables/disables the login form in the auto-generated Swagger documentation.
 The Swagger login uses the client specified in the property `springdoc.swagger-ui.oauth.client-id`.
 
 ==== Cdq
 
-The file `application-cdq.properties` enables and configures the connection to a remote https://www.apimatic.io/apidocs/data-exchange/v/4_0#/rest/getting-started[CDQ API] with which the application can exchange business partner information.
+The file `application-cdq.properties` enables and configures the connection to a remote https://www.apimatic.io/apidocs/data-exchange/v/4_0#/rest/getting-started[CDQ API] form which the application can import business partner records.
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc]: fixed

File: README.adoc [hunk 8]
```diff
@@ -3,284 +3,414 @@
...
-By enabling the corresponding properties component packages can be included in the component scan.
-
-== Kubernetes Deployment
-
-This repository contains Docker and Helm files for deploying the application to a Kubernetes environment.
-In order to deploy the application to a Kubernetes Cluster you need to containerize the application, push the resulting image to a container registry and deploy a Helm release on the prepared cluster.
-
-=== Prerequisites
+This will install a new release of the BPDM Pool in the given namespace.
+On default values this release deploys the latest image tagged as `main` from the repository's GitHub Container Registry.
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc:107]: True, I adapted the readme and will put in a fix on the default Helm charts in a separate pull request

File: README.adoc [hunk 9]
```diff
@@ -3,284 +3,414 @@
...
+applicationSecrets:
+    spring:
+      datasource:
+        password: your_database_secret
+----
+
+==== Helm Dependencies
+
+On default, the Helm deployment also contains a PostgreSQL and Opensearch deployment.
+You can configure the these deployments in your value file as well.
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc]: fixed

File: README.adoc [hunk 10]
```diff
@@ -3,284 +3,414 @@
...
-The file `application-cdq.properties` enables and configures the connection to a remote https://www.apimatic.io/apidocs/data-exchange/v/4_0#/rest/getting-started[CDQ API] with which the application can exchange business partner information.
+The file `application-cdq.properties` enables and configures the connection to a remote https://www.apimatic.io/apidocs/data-exchange/v/4_0#/rest/getting-started[CDQ API] form which the application can import business partner records.
 Depending on whether this component is enabled, the application offers endpoints to import records from and to export Business Partner Numbers to CDQ.
 When enabled the application requires the environment variable `BPDM_CDQ_KEY` to contain an API key with necessary privileges for accessing the specified storage.
+Further, you need to provide a CDQ storage ID (`BPDM_CDQ_STORAGE`) and datasource ID `BPDM_CDQ_DATASOURCE` from where the records should be imported by the application.
+
+=== Helm Deployment
 
-==== OpenSearch
+This repository contains a Helm files for deploying the BPDM Pool to a Kubernetes environment.
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply README.adoc]: fixed

File: README.adoc (hunk 1 above)
Comment (nicoprow) [reply README.adoc]: fixed

File: README.adoc (hunk 2 above)
Comment (nicoprow) [reply README.adoc]: fixed

File: README.adoc (hunk 3 above)
Comment (nicoprow) [reply README.adoc]: fixed

**Review by pazepaze — APPROVED** [review]

**Pull request merged**